runtime/compiler/control/J9Recompilation.cpp

```diff
@@ -779,4 +772,98 @@ J9::Recompilation::resetPersistentProfileInfo(TR_PersistentMethodInfo *methodInf
    methodInfo->_recentProfileInfo = NULL;
    methodInfo->_bestProfileInfo = NULL;
    }
+
+/**
+ * Serialize a TR_PersistentProfileInfo struct
+ *
+ * @param [in] profileInfo ref to TR_PersistentProfileInfo to be serialized
+ * @param [out] profileInfoStr ref to string which points to the serialized TR_PersistentProfileInfo
+ */
+void
+J9::Recompilation::serializePersistentProfileInfo(TR_PersistentProfileInfo &profileInfo, std::string &profileInfoStr)
+   {
+   uint32_t size = profileInfo.getSizeForSerialization();
+   uint8_t * buffer = (uint8_t *)TR_Memory::jitPersistentAlloc(size);
+   uint8_t * origBuffer = buffer; // buffer gets updated after the serialization
+   profileInfo.serialize(buffer);
+   profileInfoStr.append((char *)origBuffer, size);
+   TR_Memory::jitPersistentFree(origBuffer);
+   }
+
+/**
+ * Serialize the "recent" and "best" profile information stored in TR_PersistentMethodInfo
+ *
+ * @param [in] methodInfo pointer to TR_PersistentMethodInfo which stores "recent" and "best" profile info
+ * @param [out] recentProfileInfoStr ref to string which points to the serialized recent TR_PersistentProfileInfo
+ * @param [out] bestProfileInfoStr ref to string which points to the serialized best TR_PersistentProfileInfo
+ */
+void
+J9::Recompilation::serializePersistentProfileInfo(TR_PersistentMethodInfo *methodInfo, std::string &recentProfileInfoStr, std::string &bestProfileInfoStr)
+   {
+   TR_PersistentProfileInfo *recentProfileInfo = methodInfo->getRecentProfileInfo();
+   TR_PersistentProfileInfo *bestProfileInfo = methodInfo->getBestProfileInfo();
+   if (NULL != recentProfileInfo)
+      {
+      J9::Recompilation::serializePersistentProfileInfo(*recentProfileInfo, recentProfileInfoStr);
+      TR_PersistentProfileInfo::decRefCount(recentProfileInfo);
```

Comment:
This is because the above statement `methodInfo->getRecentProfileInfo();` would increment the ref count for the recent profileInfo, so on exit we decrement it.